src/__tests__/test-utils.test.tsx

```diff
@@ -0,0 +1,123 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { ComponentType } from "react";
+import { render } from "@testing-library/react";
+import { paramCase, pascalCase } from "change-case";
+import { describe, expect, test } from "vitest";
+
+import * as components from "../../lib/components";
+import createWrapperDom, { ElementWrapper as DomElementWrapper } from "../../lib/components/test-utils/dom";
+import createWrapperSelectors from "../../lib/components/test-utils/selectors";
+import { defaultProps } from "./default-props";
+
+const RENDER_COMPONENTS_DEFAULT_PROPS: Record<string, unknown>[] = [
+  {
+    "data-testid": "first-item",
+    "data-name": "first item",
+  },
+  {
+    "data-testid": "second-item",
+    "data-name": "second item",
+  },
+];
+
+function renderComponents(componentName: string, props = RENDER_COMPONENTS_DEFAULT_PROPS) {
+  const Component = components[componentName as keyof typeof components] as ComponentType;
+  const componentDefaultProps = defaultProps[paramCase(componentName) as keyof typeof defaultProps];
+  return render(
+    <div>
+      {props.map(({ ...customProps }, index) => (
+        <Component key={index} {...componentDefaultProps} {...customProps} />
+      ))}
+    </div>,
+  );
+}
+
+function getComponentSelectors(componentName: string) {
+  const componentNamePascalCase = pascalCase(componentName);
+  const findAllRegex = new RegExp(`findAll${componentNamePascalCase}.*`);
+
+  // The same set of selector functions are present in both dom and selectors.
+  // For this reason, looking into DOM is representative of both groups.
+  const wrapperPropsList = Object.keys(DomElementWrapper.prototype);
+
+  // Every component has the same set of selector functions.
+  // For this reason, casting the function names into the Alert component.
+  const findName = `find${componentNamePascalCase}` as "findCodeView";
+  const findAllName = wrapperPropsList.find((selector) => findAllRegex.test(selector)) as "findAllCodeViews";
```

Comment:
Why here the type of findName and findAllName `as "findCodeView"`? Can it be other component?

Reply:
This is for simplification. All of the `findAllXX` and `findXX` behave exactly the same. So casting it into one of them is enough for this test. Otherwise there's nothing special about `CodeViews`. It can be any other component from this repo.

Reply:
Also fixed the comment above ✅